master/internal/api_checkpoint.go

```diff
@@ -28,6 +37,59 @@ func (a *apiServer) GetCheckpoint(
 	}
 }
 
+func (a *apiServer) DeleteCheckpoints(
+	ctx context.Context,
+	req *apiv1.DeleteCheckpointsRequest) (*apiv1.DeleteCheckpointsResponse, error) {
+	deleteCheckpoints := req.CheckpointUuids
+	dCheckpointsInModelRegistry, _ := a.m.db.GetDeleteCheckpointsInModelRegistry(deleteCheckpoints)
+	// return 400 if model registry checkpoints and include all the model registry checkpoints
+
+	if len(dCheckpointsInModelRegistry) > 0 {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"This subset of list of checkpoints provided are in the model registry: %v cannot be deleted.",
+			dCheckpointsInModelRegistry)
+	}
+
+	jsonVCheckpoints, _ := json.Marshal(deleteCheckpoints)
+	taskSpec := *a.m.taskSpec
```

Comment:
we will need to fill in the agent user group (see https://github.com/determined-ai/determined/blob/master/master/internal/api_command.go#L64-L75) else this may break for shared_fs checkpoints (there are a few other ways, too)